General.ruleset

```diff
@@ -0,0 +1,110 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="General C# rules" Description="" ToolsVersion="16.0">
```

Comment:
No configuration here, as well. I know you don't follow this ordering system of class members, but I would consider it one more time, as this is the commonly used ordering.

Reply:
So disable it?

Reply:
I'm still working on this and the other open questions to see what we want with the team, and how a custom analyzer is possible.

Reply:
I've disabled this.

Reply:
There is configurability for this actually, see: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md#element-order So please do configure it according to our documentation.

Reply:
Cannot find your docs for this. Could you please compose an ordered list of these values: "kind", "accessibility", "constant", "static", "readonly" that you think suits your standards? I can then add this configuration to `stylecop.json`. I also guess that "kind" should be excluded, as you follow another (comparing to [SA1201](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md)) class members order.

Reply:
See [here](https://orcharddojo.net/orchard-resources/CoreLibrary/DevelopmentGuidelines/CodeStyling) under "Have a standard ordering of members..." For everything else not indicated there please just use the StyleCop default and we'll fine tune it later.

Reply:
OK, we have a problem here. Orchard standards look completely incompatible with the ones that StyleCop provides:

- `kind` (elements are ordered according to their kind (see [SA1201](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md) for this predefined order)) - Orchard class members order differs from the order defined in SA1201
- `accessibility` (elements are ordered according to their declared accessibility (see [SA1202](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1202.md) for this predefined order)) - valid for Orchard methods but not for fields: *Notice the order: static, private, protected, public, const, constructor, public, protected, private, static, inner classes.*
- `constant` (constant elements are ordered before non-constant elements) - in Orchard vice versa: *Constants just before the constructor* (means after all other fields).
- `static` (static elements are ordered before non-static elements) - in Orchard - static fields before and static methods after the rest.
- `readonly` (readonly elements are ordered before non-readonly elements) - in Orchard style I don't see anything about `readonly`.

So if you continue to follow Orchard's style, from this configuration only `readonly` probably can apply. And I don't see much advantage of this.

Reply:
Oh damn.

I think the following ones would work though:

- accessibility
- constant (now that I see it we're actually doing what the rule says not what's documented)
- readonly

Please enable these and I'll check what happens, and adjust if necessary. We'd need [more configuration](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/1610), unfortunately.

Reply:
Configured it this way. Please check it out.

Reply:
Looks good. There are no SA1201 violations actually, just SA1202 ones. There are a lot of mismatches for accessibility though indeed, so that won't be usable. I've thus disabled that one.